### ISSUE TITLE
Declaration of GenericLanguageInflectorFactory is not compatible with interface

### ISSUE THREAD

**Issue by NotionCommotion** [rectorphp/rector#6556]
I added rector to a Symfony project and got this error.  To eliminate as much uncertainty as possible, created another project with just rector and am getting the same error.  Any suggestions?  Thank you

```
[michael@devserver test3]$ ls -la
total 4
drwxr-xr-x  2 michael michael    6 Jul  5 16:06 .
drwxr-xr-x 39 root    root    4096 Jul  5 16:06 ..
[michael@devserver test3]$ composer require rector/rector --dev
Using version ^0.11.34 for rector/rector
./composer.json has been created
Running composer update rector/rector
Loading composer repositories with package information
Updating dependencies
Lock file operations: 2 installs, 0 updates, 0 removals
  - Locking phpstan/phpstan (0.12.90)
  - Locking rector/rector (0.11.34)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 2 installs, 0 updates, 0 removals
  - Installing phpstan/phpstan (0.12.90): Extracting archive
  - Installing rector/rector (0.11.34): Extracting archive
Generating autoload files
2 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
[michael@devserver test3]$ vendor/bin/rector init
PHP Fatal error:  Declaration of RectorPrefix20210704\Doctrine\Inflector\GenericLanguageInflectorFactory::withSingularRules(?RectorPrefix20210704\Doctrine\Inflector\Rules\Ruleset $singularRules, bool $reset = false): RectorPrefix20210704\Doctrine\Inflector\GenericLanguageInflectorFactory must be compatible with RectorPrefix20210704\Doctrine\Inflector\LanguageInflectorFactory::withSingularRules($singularRules, bool $reset = false) in /var/www/test3/vendor/rector/rector/vendor/doctrine/inflector/lib/Doctrine/Inflector/GenericLanguageInflectorFactory.php on line 23
Fatal error: Declaration of RectorPrefix20210704\Doctrine\Inflector\GenericLanguageInflectorFactory::withSingularRules(?RectorPrefix20210704\Doctrine\Inflector\Rules\Ruleset $singularRules, bool $reset = false): RectorPrefix20210704\Doctrine\Inflector\GenericLanguageInflectorFactory must be compatible with RectorPrefix20210704\Doctrine\Inflector\LanguageInflectorFactory::withSingularRules($singularRules, bool $reset = false) in /var/www/test3/vendor/rector/rector/vendor/doctrine/inflector/lib/Doctrine/Inflector/GenericLanguageInflectorFactory.php on line 23
[michael@devserver test3]$ php -v
PHP 8.0.8 (cli) (built: Jun 29 2021 07:41:19) ( NTS gcc x86_64 )
Copyright (c) The PHP Group
Zend Engine v4.0.8, Copyright (c) Zend Technologies
[michael@devserver test3]$ composer -V
Composer version 2.0.8 2020-12-03 17:20:38
[michael@devserver test3]$ cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

[michael@devserver test3]$
```

**Comment by samsonasik** [comment]
You can temporary pin the exact version "rector/rector": "0.11.32" in composer.json. Duplicate with https://github.com/rectorphp/rector/issues/6543

**Comment by NotionCommotion** [comment]
Thank you samsonasik,  Ended up doing the following before realizing you responded. ` Now getting PHP Fatal error:  Uncaught Error: Call to undefined function normalizer_is_normalized() in /var/www/test/vendor/rector/rector/vendor/symfony/string/UnicodeString.php:35`  and hoping that 0.11.32 will also help.


```
diff --git a/vendor/rector/rector/vendor/doctrine/inflector/lib/Doctrine/Inflector/LanguageInflectorFactory.php b/vendor/rector/rector/vendor/doctrine/inflector/lib/Doctrine/Inflector/LanguageInflectorFactory.php
index b79f488..d68b692 100644
--- a/vendor/rector/rector/vendor/doctrine/inflector/lib/Doctrine/Inflector/LanguageInflectorFactory.php
+++ b/vendor/rector/rector/vendor/doctrine/inflector/lib/Doctrine/Inflector/LanguageInflectorFactory.php
@@ -14,7 +14,7 @@ interface LanguageInflectorFactory
      * @return $this
      * @param \Doctrine\Inflector\Rules\Ruleset|null $singularRules
      */
-    public function withSingularRules($singularRules, bool $reset = \false);
+    public function withSingularRules(?Ruleset $singularRules, bool $reset = \false);
     /**
      * Applies custom rules for pluralisation
      *
@@ -23,7 +23,7 @@ interface LanguageInflectorFactory
      * @return $this
      * @param \Doctrine\Inflector\Rules\Ruleset|null $pluralRules
      */
-    public function withPluralRules($pluralRules, bool $reset = \false);
+    public function withPluralRules(?Ruleset $pluralRules, bool $reset = \false);
     /**
      * Builds the inflector instance with all applicable rules
      */
diff --git a/vendor/rector/rector/vendor/symfony/console/Helper/HelperInterface.php b/vendor/rector/rector/vendor/symfony/console/Helper/HelperInterface.php
index a6f3260..5d60306 100644
--- a/vendor/rector/rector/vendor/symfony/console/Helper/HelperInterface.php
+++ b/vendor/rector/rector/vendor/symfony/console/Helper/HelperInterface.php
@@ -21,7 +21,7 @@ interface HelperInterface
      * Sets the helper set associated with this helper.
      * @param \Symfony\Component\Console\Helper\HelperSet|null $helperSet
      */
-    public function setHelperSet($helperSet = null);
+    public function setHelperSet(\RectorPrefix20210704\Symfony\Component\Console\Helper\HelperSet $helperSet = null);
     /**
      * Gets the helper set associated with this helper.
      *
diff --git a/vendor/rector/rector/vendor/symfony/console/Output/Output.php b/vendor/rector/rector/vendor/symfony/console/Output/Output.php
index 248e5a5..b12b6ad 100644
--- a/vendor/rector/rector/vendor/symfony/console/Output/Output.php
+++ b/vendor/rector/rector/vendor/symfony/console/Output/Output.php
@@ -151,5 +151,5 @@ abstract class Output implements \RectorPrefix20210704\Symfony\Component\Console
      * @param string $message
      * @param bool $newline
      */
-    protected abstract function doWrite($message, $newline);
+    protected abstract function doWrite(string $message, bool $newline);
 }

```

**Comment by NotionCommotion** [comment]
Above fix is no longer needed as 0.11.35 addresses the issue.
